eliHenderson/assignments/slidingwindow/lc53/lc53.js

```diff
@@ -1,5 +1,40 @@
+/**
+ * Given the int arr num, find cont sub arr containing at
+ *  least one number which has the largest sum and return its sum;
+ *
+ * My version: Array, Int
+ * find: largest sub array sum
+ * return : largest sum
+ *
+ * there can be neg numbers in the array
+ *
+ * edge: for 0's  return 0
+ * edge: [] array should have at least one elem
+ *
+ */
```

Comment:
High level overview needs a little bit of work.

Example:
these are my edges I can think of, X Y Z
these are my constraints, only positive integers input array and only positive integer K
I will attempt to solve this using a sliding window approach
init maxSum, tempSum & windowStart
windowStart will begin at 0
init a for loop, use windowEnd naming convention as opposed to traditional i 
add each element in the array to my tempsum until windowend is >= K
at that point I want to start comparing against and reassigning my max sum
right after this I want to shrink my window by removing the element @ windowStart from temp and incrementing windowStart so that it moves to the next index
the for loop will iterate and my window end will grow bringing a new size K window into view
the program will continue comparing and reassigning maxSum until we run out of elements in the array
lastly I will return my maxSum